### PR TITLE
Blasters Update 2

### PR DESCRIPTION
**Pull request by kwazedilla**
blaster update 50
(This WILL be the update banner or the PR isn't going through!)

New multiblock: **Ammo Press**
- Refills blasters and magazines placed in the furnace
- Consumes power (250) and refill items (placed in the rear chest)

New magazine type: **Special Magazine**
- 20 ammo
- Refill with Emeralds (20 ammo per item)
- Blaster Snipers and Blaster Shotguns will now reload from Special Magazines

Updated magazine type: **Standard Magazine**
- Refill with Lapis Lazuli (20 ammo per item)
- Blaster Rifles, Blaster Pistols, and Submachine Blasters will now reload from Standard Magazines

New sounds: **Blasters**
- All blasters now have unique sounds for firing, and reloading
- Firing sounds will sound more distant if the player is further away
- New hit marker sound
- Projectile whizz and impact sounds
- Sounds are muted in space

Blaster GUI & utility changes
- Refreshed ammo interface in item lore; added refill item and magazine (if applicable) to lore
- Durability bar now shows remaining ammunition for blasters and magazines
- Magazines of the same type may be crafted together to form a single magazine with the combined ammo

Other & Bug fixes
- `/customitems` will now post an error if an incorrect item is given, or the quantity is less than 1
- Blasters will no longer take durability damage
- Rocket boots are now correctly disabled when shot
- Projectiles that can penetrate and headshot will no longer despawn if a headshot is scored
- Projectiles that hit a player with active rocket boots will now correctly play a hitmarker
- Blaster balancing configuration default values has been changed to match the live server

### CODE REVIEW

File: server/src/main/kotlin/net/horizonsend/ion/server/configuration/BalancingConfiguration.kt
Comment (Gutin1) [inline server/src/main/kotlin/net/horizonsend/ion/server/configuration/BalancingConfiguration.kt:1]: If the magazine type is specified and has a refill type, having it here is redundant

File: server/src/main/kotlin/net/horizonsend/ion/server/features/blasters/BlasterListeners.kt
Comment (Gutin1) [inline server/src/main/kotlin/net/horizonsend/ion/server/features/blasters/BlasterListeners.kt:1]: Might be faster to use player cache

File: server/src/main/kotlin/net/horizonsend/ion/server/configuration/BalancingConfiguration.kt
Comment (kwazedilla) [reply server/src/main/kotlin/net/horizonsend/ion/server/configuration/BalancingConfiguration.kt:1]: Specifying the refill type in addition to the magazine time allows for blasters to be reloaded via ammo press.

Comment (kwazedilla) [reply server/src/main/kotlin/net/horizonsend/ion/server/configuration/BalancingConfiguration.kt:1]: Resolved; blaster refill type removed; now obtained by accessing magazine type's refill type

**Pull request merged**